src/Context/data-context.js

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext, useReducer } from "react";
+import { dataReducer } from "../Reducer/data-reducer";
+
+export const DataContext = createContext();
+export const useDataContext = () => useContext(DataContext);
+
+const initialArg = {
+    videoList:[],
+    likedVideos:[],
+    history:[],
+    playlist:[
+      {
+        listId: 1,
```

Comment:
I have a suggestion here, you can change "listId" with "id", as when we declare the array is of playlists, we can say id. If you want you can keep it this way, anyways this will get refactored while doing backend.

Reply:
yes, I shall change this while developing the backend!